src/dmd/parse.d

```diff
@@ -4973,19 +4989,37 @@ class Parser(AST) : Lexer
             }
             else if (ident)
             {
+                AST.Expression width;
+                if (token.value == TOK.colon)
+                {
+                    nextToken();
+                    width = parseCondExp();
+                }
+
                 AST.Initializer _init = null;
                 if (token.value == TOK.assign)
                 {
                     nextToken();
                     _init = parseInitializer();
                 }
 
-                auto v = new AST.VarDeclaration(loc, t, ident, _init);
-                v.storage_class = storage_class;
-                if (pAttrs)
-                    pAttrs.storageClass = STC.undefined_;
-
-                AST.Dsymbol s = v;
+                AST.Dsymbol s;
+                if (width)
+                {
+                    if (_init)
+                        error("no initializer for bit-field declaration");
+                    if (storage_class)
+                        error("no storage class for bit-field declaration");
```

Comment:
These error messages are confusing to me. It suggests the programmer forgot to give an initializer / storage class, while they should tell that they are *not allowed* on bit fields.

Reply:
For better error messages, I encourage you to submit a followup PR for it.

Reply:
changed it.